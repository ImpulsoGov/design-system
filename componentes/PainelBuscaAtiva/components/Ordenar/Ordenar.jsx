import React from "react";
import { ButtonColorSubmit } from "../../../ButtonColor";
import { ButtonLightSubmit } from "../../../ButtonLight";
import * as helpers from "../../helpers";
import style from "./Ordenar.module.css";
import { CardFiltro } from "../CardFiltro";

export const Ordenar = ({
  rotulosfiltros,
  IDFiltros,
  filtros,
  setChavesFiltros,
  dadosFiltros,
  tabela,
  setOrdenar,
  data,
  ordenar,
  datefiltros,
  IntFiltros,
  setModal,
  setOrdenacaoAplicada,
  IDFiltrosOrdenacao,
  trackObject,
  painel,
  aba,
  sub_aba,
  setShowSnackBar,
  updateData,
})=>{
  const filtros_painel = {
    "rotulos" : rotulosfiltros,
    "ID" : IDFiltros
  }
  const hasFiltersApplied = Object.values(filtros).some((value) => value);

  function filter(data) {
    const chosenFilters = helpers.valuesToChavesFiltros(filtros, setChavesFiltros, dadosFiltros);
    const groupedFilters = helpers.agruparChavesIguais(chosenFilters);

    return helpers.filterByChoices(data, groupedFilters);
  }

  const limparOrdenacao = ()=>{
    const dados = hasFiltersApplied ? filter(tabela) : tabela

    setOrdenar()
    updateData(dados)
    setModal(false)
    setOrdenacaoAplicada(false)
  }

  const handleSortClick = ()=>{
    const dados = hasFiltersApplied ? filter(data) : data

    updateData(helpers.sortByChoice(dados, ordenar, IDFiltrosOrdenacao, datefiltros, IntFiltros))

    setShowSnackBar({
      open: true,
      message: "Lista ordenada com sucesso!",
      background: "#2EB280",
      color: "white",
    })

    trackObject.track('button_click', {
      'button_action': 'aplicar_ordenacao',
      'nome_lista_nominal': painel,
      'aba_lista_nominal' : aba,
      'sub_aba_lista_nominal' : sub_aba,
      'button_choices' : ordenar
    })
    setModal(false)
    setOrdenacaoAplicada(true)
  }

  return(
    <div className={style.containerOrdenar} data-testid="Ordenar">
      <p className={style.OrdenarPor}>Ordenar por:</p>

      {filtros_painel.rotulos.map((label)=> (
        <CardFiltro
          label={label}
          isSelected={ordenar === filtros_painel.ID[label]}
          sortProperty={filtros_painel.ID[label]}
          setOrdenar={setOrdenar}
          key={label}
        />
      ))}

      <div className={style.AplicarOrdenacao}>
        <ButtonLightSubmit
          label="Limpar ordenação"
          submit={limparOrdenacao}
        />

        <ButtonColorSubmit
          label="ORDENAR LISTA"
          submit={handleSortClick}
          arg={{}}
        />
      </div>
    </div>
  )
}