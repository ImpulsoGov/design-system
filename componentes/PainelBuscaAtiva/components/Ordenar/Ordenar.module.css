.containerOrdenar{
  padding: 5px 40px;
  display: flex;
  flex-direction: column;
  gap:15px;
}

.limparOrdenacao{
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 150%;
  color: #2EB380;
  border-bottom: #2EB380 solid 1.5px;
  width: fit-content;
}

.limparOrdenacao:hover{
  cursor: pointer;
}

.OrdenarPor{
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 130%;
  letter-spacing: -0.02em;
  color: #1F1F1F;
}

