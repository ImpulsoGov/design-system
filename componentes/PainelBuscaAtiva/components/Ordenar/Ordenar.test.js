import { render, screen } from '@testing-library/react';
import { Ordenar } from './Ordenar';

const COMPONENT = 'Ordenar';
const scenarios = [
  {
    painel: 'citopatológico',
    aba: 1,
    sub_aba: 0,
    ordenacao: '1',
    setOrdenar: jest.fn(),
    data: [],
    setData: jest.fn(),
    tabela: [
      {
        paciente_nome: 'Camila da Silva',
        cidadao_cpf_dt_nascimento: '106.106.106-10',
        id_status_usuario: 12,
        vencimento_da_coleta: '-',
        prazo_proxima_coleta: '31/08/2023',
        idade: 25,
        id_faixa_etaria: 8,
        acs_nome: 'Alessandra Santos',
        estabelecimento_cnes: '2872872',
        estabelecimento_nome: 'Unidade de Saude da Familia 1',
        equipe_ine: '0002277227',
        ine_master: '0002277227',
        equipe_nome: 'ESF 1',
        dt_registro_producao_mais_recente: '2023-10-22'
      },
      {
        paciente_nome: 'Maria da Silva',
        cidadao_cpf_dt_nascimento: '327.327.327-32',
        id_status_usuario: 12,
        vencimento_da_coleta: '11/05/2026',
        prazo_proxima_coleta: '31/08/2023',
        idade: 35,
        id_faixa_etaria: 6,
        acs_nome: 'Alessandra Santos',
        estabelecimento_cnes: '2872872',
        estabelecimento_nome: 'Unidade de Saude da Familia 1',
        equipe_ine: '0002277227',
        ine_master: '0002277227',
        equipe_nome: 'ESF 1',
        dt_registro_producao_mais_recente: '2023-10-22'
      },
      {
        paciente_nome: 'Julia da Silva',
        cidadao_cpf_dt_nascimento: '100.100.100-10',
        id_status_usuario: 12,
        vencimento_da_coleta: '27/07/2025',
        prazo_proxima_coleta: 'Em dia',
        idade: 55,
        id_faixa_etaria: 8,
        acs_nome: 'Carmen Miranda',
        estabelecimento_cnes: '2752752',
        estabelecimento_nome: 'Unidade de Saude da Familia 2',
        equipe_ine: '0000369369',
        ine_master: '0000369369',
        equipe_nome: 'ESF 2',
        dt_registro_producao_mais_recente: '2023-10-22'
      },
      {
        paciente_nome: 'Carla da Silva',
        cidadao_cpf_dt_nascimento: '305.305.305-30',
        id_status_usuario: 13,
        vencimento_da_coleta: '-',
        prazo_proxima_coleta: '31/08/2023',
        idade: 64,
        id_faixa_etaria: 8,
        acs_nome: 'Alessandra Santos',
        estabelecimento_cnes: '2872872',
        estabelecimento_nome: 'Unidade de Saude da Familia 1',
        equipe_ine: '0002277227',
        ine_master: '0002277227',
        equipe_nome: 'ESF 1',
        dt_registro_producao_mais_recente: '2023-10-22'
      },
    ],
    datefiltros: ['vencimento_da_coleta'],
    IntFiltros: ['idade'],
    setModal: jest.fn(),
    setOrdenacaoAplicada: jest.fn(),
    IDFiltros: {
      "NOMES DE A-Z": "paciente_nome",
      "IDADE": "idade",
      "VENCIMENTO DA COLETA MAIS ANTIGO": "vencimento_da_coleta",
    },
    rotulosfiltros: [
      "NOMES DE A-Z",
      "VENCIMENTO DA COLETA MAIS ANTIGO",
      "IDADE",
    ],
    IDFiltrosOrdenacao: {
      "paciente_nome": "asc",
      "idade": "asc",
      "vencimento_da_coleta": "desc",
    },
    trackObject: { track: jest.fn() },
    filtros: {
      'Alessandra Santos': false,
    },
    setChavesFiltros: jest.fn(),
    dadosFiltros: [
      {
        data: ['Alessandra Santos'],
        filtro: 'acs_nome',
        rotulo: 'Filtrar por nome do Profissional Responsável'
      },
    ],
  }
];

describe(`Componente: ${COMPONENT}`, () => {
  it('deve renderizar corretamente', () => {
    render(
      <Ordenar { ...scenarios[0] } />
    );
    const component = screen.getByTestId(COMPONENT);
    expect(component).toMatchSnapshot();
  });
});