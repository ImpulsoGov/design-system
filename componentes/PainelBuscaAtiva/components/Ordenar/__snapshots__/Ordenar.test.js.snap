// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente: Ordenar deve renderizar corretamente 1`] = `
<div
  class="containerOrdenar"
  data-testid="Ordenar"
>
  <div
    class="limparOrdenacao"
  >
    Limpar ordenação
  </div>
  <p
    class="OrdenarPor"
  >
    Ordenar por:
  </p>
  <div
    class="cardFiltro"
    data-testid="CardFiltro"
  >
    NOMES DE A-Z
  </div>
  <div
    class="cardFiltro"
    data-testid="CardFiltro"
  >
    VENCIMENTO DA COLETA MAIS ANTIGO
  </div>
  <div
    class="cardFiltro"
    data-testid="CardFiltro"
  >
    IDADE
  </div>
  <button
    class="ButtonColorContainer ColorIP"
  >
    ORDENAR LISTA
  </button>
</div>
`;
